Update stable memory test readme (#1038)
dfinity/internet-identity: src/internet_identity/stable_memory/README.md

# Stable Memory Tests

> These test serve two purposes:

Reviewer: ```suggestion
These tests serve two purposes:
```

* Ensure backward compatibility with our stable memory format.
* Show that we can indeed fully recover in case of a disaster, assuming
  we have a backup of the stable memory.

## Creating New Test Memory Backups

### Using Canister Test Infrastructure
These backup memory files are generated using the `canister_tests` infrastructure. To create a new backup file:
1. Install II on the `StateMachine`
    ```rust
    let env = StateMachine::new();
    let canister_id = install_ii_canister(&env, framework::II_WASM.clone());
    ```
2. Create the desired state in II by interacting with it.
3. Export the stable memory using the helper function provided by the `framework`:
    ```rust
    framework::save_compressed_stable_memory(&env, canister_id, "stable_memory/stable.bin.gz", "stable_memory/stable.bin.gz");
    ```
### Using the Local Replica

The stable memory can also be copied from the local replica from the following path:

> `~/Library/Application Support/org.dfinity.dfx/network/local/state/replicated_state/node-100/state/tip/canister_states/CANISTER_ID/stable_memory.bin`

Reviewer: Is there some `XDG_FOO` we can use here? This won't work for Linux users

## Registering the DFX Key as II Device

Registering the `dfx` public key in II can be useful for testing as certain calls (e.g. `register`) require the `caller` to match the submitted public keys.

To register dfx as a device:
1. Export the key in the DER file format:
    ```bash
    dfx identity export IDENTITY | openssl ec -out FILE_NAME -pubout -outform der
    ```
2. Make sure that the current `dfx` identity matches the one exported in step 1
    ```bash
    dfx identity use IDENTITY
    ```

> 3. Register an anchor using `dfx` (assuming DUMMY_CAPTCHA, otherwise the actual challenge key has to be submitted)

Reviewer: ```suggestion
3. Register an anchor using `dfx` (assuming DUMMY_CAPTCHA, otherwise the actual challenge `chars` have to be submitted)
```
